Integrated Google Calendar on the Events page

shravani05 opened the pull request:
### Description

Integrated Google Calendar on the Events page. 

Fixes #121 

### Type of Change:

- Code
- User Interface


**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

![image](https://user-images.githubusercontent.com/11148726/102816282-aa513880-43c5-11eb-9fb2-d2774b70b837.png)

This can be tested here https://shravani05.github.io/anitab-org.github.io/

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have attached link of deployed site. https://shravani05.github.io/anitab-org.github.io/
- [ ] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

Bucky25 commented:
@shravani05 please add a link of deployed gh-page

shravani05 commented:
> @shravani05 please add a link of deployed gh-page

@Bucky25  Here is the deployed link: https://shravani05.github.io/anitab-org.github.io/
I have also attached the image of the Events page
![Cal](https://user-images.githubusercontent.com/43898702/101483713-94cc1f80-397e-11eb-9d4b-0c1323bc7a8d.png)

tichnas commented:
@shravani05 Thanks for this PR :)
Also, it would be great if you change the inline styles to react-native Stylesheet

shravani05 commented:
> @shravani05 Thanks for this PR :)
> Also, it would be great if you change the inline styles to react-native Stylesheet

@tichnas I have changed the styles to react-native StyleSheet for all the components. But the iframe component can't receive style created by StyleSheet.create(), so I have kept inline style for it.

Bucky25 commented on src/Components/Events/GoogleCalendar/index.js:
```diff
@@ -1,22 +1,24 @@
 
-export default GoogleCalendar;
\ No newline at end of file
+const styles = StyleSheet.create({
+  container: {
+    width: "80%",
+    alignItems: 'left',
+    paddingLeft: 16,
+    paddingRight: 16,
+  }
+});
```
@shravani05 please add some padding between above para and calendar heading

Bucky25 commented on src/Components/Events/index.js:
```diff
@@ -1,23 +1,12 @@
-      }}>EVENT CALENDAR</Text> 
-      <Text style={{color: "#013287"}}>
+    <View style={styles.container}>
+      <Text h1 style={styles.heading}>EVENT CALENDAR</Text> 
```
please sectionsubheader to write sub heading

Bucky25 reviewed (changes requested):
please use design give to add padding or margin between components

shravani05 commented:
@Bucky25 please let me know if there are any other changes to be made 
https://shravani05.github.io/anitab-org.github.io/
![Cal1](https://user-images.githubusercontent.com/43898702/101942567-bf80d700-3c0f-11eb-8ce0-80211895b51a.png)

Bucky25 commented on src/Components/Events/index.js:
```diff
@@ -0,0 +1,37 @@
+  return (  
+      <View style={styles.container}>
+      <SectionSubheader title="Event Calendar"/>
+      <Text style={styles.data}>
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer malesuada massa nec porta interdum. 
+      Vivamus porta mollis urna at congue. 
+      Maecenas mollis, dolor ut auctor ultrices, diam ex dignissim ante, at sollicitudin risus risus id felis. 
+      Suspendisse vel ante placerat, feugiat ligula pretium, dignissim est. 
+      Donec vestibulum ullamcorper sapien et egestas. Ut eu aliquet leo.
+      </Text>
```
@shravani05  please style this text like other pages having text below heading

shravani05 commented:
@Bucky25 I have added styles as per other pages. Please let me know if any further changes are required.

isabelcosta commented on src/Components/Events/GoogleCalendar/index.js:
```diff
@@ -0,0 +1,15 @@
+  return (
+    <MainContainer>
+        {/* Replace this iframe with the iframe of the google calendar associated with the google account according to which you want the schedule */}
+        <iframe src="https://calendar.google.com/calendar/embed?src=er3gi62a5hip5jnk1f7r0al62s%40group.calendar.google.com&ctz=America%2FLos_Angeles" 
```
```suggestion
        <iframe src="https://calendar.google.com/calendar/embed?src=er3gi62a5hip5jnk1f7r0al62s%40group.calendar.google.com" 
```
I think if we don't put the `ctz` query parameter, the  timezone assumed will be the one from the computer accessing the page.

isabelcosta commented on src/Components/Events/index.js:
```diff
@@ -0,0 +1,33 @@
+        <Content>
+          <SectionSubheader title="Event Calendar"/>
+          <Description>
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer malesuada massa nec porta interdum. 
```
Not sure about the content yet, but you could mention the external links to the calendar in different timezones. Here's the text you can put:
```suggestion
          We host multiple open sessions to the community where members of the community can discuss concerns about the community, ask questions, discuss projects, etc. These sessions are open to anyone and are usually conducted via Zoom. These are audio-only calls, so no video is required.
          You can see our calendar in different timezones:
```

Then in front of ":" you could add HTML code to link each timezone to its calendar like: ([PST](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=America/Los_Angeles) || [EST](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=America/New_York) || [GMT](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=GMT) || [WAT](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=Africa/Lagos) || [IST](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=Asia/Colombo) || [AEDT](https://calendar.google.com/calendar/embed?src=sh10tv3mtfve62somg9nngp9tg%40group.calendar.google.com&ctz=Australia/Sydney))

@shravani05

shravani05 replied:
@isabelcosta On it!

isabelcosta reviewed (changes requested):
This is looking amazing, I would like to suggest a small improvement. Please read my comment about the calendar in different timezones, and let me know if you need any clarification :) @shravani05 

Thank you so much for your contribution!

shravani05 commented:
@isabelcosta I have added the links as mentioned. Please let me know if any further changes are required :)

isabelcosta reviewed (approved):
I tested this with the link you provided, and functionality wise this looks amazing 💯 🎉 I learned a little bit from your implementation :D

Great work @shravani05 👏 👏 👏 
Thank you so much for this contribution!

isabelcosta commented:
@shravani05 as a suggestion for your future contributions, if you do a change to user interface, add an image of the UI to the PR description and also add the link you sent on the latest comment. I just did that for you in your PR description, now it looks much more informative for whoever reviews your work :)

mayburgos commented:
@anitab-org/admins Quick note: Add another issue to see if we can research how to rename through the URL link.

tichnas reviewed (changes requested):
Delete package-lock.json file as it isn't required for yarn package manager (which we are using).
Also, instead of using `var`, use `const` or `let`.

tichnas commented on src/Components/Events/style.js:
```diff
@@ -0,0 +1,42 @@
+import { View, Text } from 'react-native';
```
Correct the indentation in this file

tichnas commented on src/Components/Events/index.js:
```diff
@@ -0,0 +1,76 @@
+        <Content>
+          <SectionSubheader title="Event Calendar"/>
+          <Description>
+          We host multiple open sessions to the community where members of the community can discuss concerns about the community, ask questions, discuss projects, etc. These sessions are open to anyone and are usually conducted via Zoom. These are audio-only calls, so no video is required.
```
Not sure, but if this paragraph isn't constant we should put it somewhere outside the code (inside content folder)
What do you say @isabelcosta?

isabelcosta replied:
agreed. I was not aware of the whole code architecture, that seems to make sense @tichnas :)

shravani05 commented:
> @shravani05 as a suggestion for your future contributions, if you do a change to user interface, add an image of the UI to the PR description and also add the link you sent on the latest comment. I just did that for you in your PR description, now it looks much more informative for whoever reviews your work :)

Thank you so much for the suggestion😊@isabelcosta 
It has been great working on this issue!

annabauza commented on src/Components/Events/GoogleCalendar/index.js:
```diff
@@ -0,0 +1,14 @@
+  return (
+    <MainContainer>
+        <iframe src={`https://calendar.google.com/calendar/u/0/embed?src=sh10tv3mtfve62somg9nngp9tg@group.calendar.google.com${timezone}`}
+          title="Calendar" style={{border: "0", width: "100%", height:"500px", frameborder:"0", scrolling:"no"}}>
```
can we move style from inline?

annabauza replied:
are pixels necessary ?

shravani05 replied:
@annabauza  I tried adding react-native StyleSheet for the iframe, but the iframe component can't receive style created by StyleSheet.create(), so I have kept inline style for it. Please let me know if there is any other way to do it.
And pixels are necessary to get the required height.

annabauza replied:
Fair enough

annabauza commented on src/Components/Events/index.js:
```diff
@@ -0,0 +1,76 @@
+        <Content>
+          <SectionSubheader title="Event Calendar"/>
+          <Description>
+          We host multiple open sessions to the community where members of the community can discuss concerns about the community, ask questions, discuss projects, etc. These sessions are open to anyone and are usually conducted via Zoom. These are audio-only calls, so no video is required.
```
Can we move this to content, please?

annabauza reviewed (changes requested):
few tweaks and it looks really good. Thank you!

shravani05 commented:
Sorry for delay in the further commits, I was busy last month with my exams. @tichnas I have replaced the var with const and also corrected the indentation in the style.js file. Also, I have moved the description to the content folder. Please let me know if any further changes are required :)
The deployed site can be found here: https://shravani05.github.io/anitab-org.github.io/

annabauza reviewed (approved):
:100:

isabelcosta commented:
@shravani05 can you please fix the merge conflicts?

shravani05 commented:
@isabelcosta I have resolved the merge conflicts. Please have a look here:  https://shravani05.github.io/anitab-org.github.io/

isabelcosta commented:
@annabauza this has 2 approvals, so it's ready for merging. Is this correct?
I checked it on @shravani05 website https://shravani05.github.io/anitab-org.github.io/ and it seems to be OK 👏 

![image](https://user-images.githubusercontent.com/11148726/104820460-aed60900-582c-11eb-8ea9-42f22824567b.png)
